### DISCUSSION TITLE
Autofill a field with flow

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#16335] [Archive]
Hello, I would like to fill a field with values if they have been left blank.

The field "Owner" is an "M2N" field with reference to one or more users. If someone is "Owner", he gets access to the record via filtered access rights. Therefore, the field must never be left empty.

If someone has not specified the "Owner", they should be entered as the "Owner" themselves. This should work both via the UI and the API, so I would do it via flow. 

Unfortunately, I can't get any further, although there are only two steps.

Step 1: Filter condition if the field was left empty.
`
{
    "$trigger": {
        "Owner": {
            "id": {
                "_null": true
            }
        }
    }
}
`

Step 2: Complete the payload with the current user.
`
{
    "Owner": {
        "create": [
            {
                "UrmVitality_id": "{{ $trigger.keys[0] }}",
                "directus_users_id": {
                    "id": "{{ $accountability.user }}"
                }
            }
        ],
        "update": [],
        "delete": []
    }
}
`

Any help or good idea?

**Comment by br41nslug** [comment]
In step 2 you're updating a relational field but it looks like you're missing the ID of the main table you're updating 😬

**Reply by ghost** [reply]
Hi ... I just got this info from the log. If I enter it manually in the UI, then this is output as a payload. So I thought this would also be what I need to adjust as payload if it is missing.

The first step does not work either, because it always return "true".

**Reply by br41nslug** [reply]
This can indeed be the update object if you're explicitly updating an item by ID `PATCH /items/:collection/:ID` (this is what the app does). To mirror this make sure that the ID of the record you're updating is present either in the object of in the IDs list of the update operation.
![image](https://user-images.githubusercontent.com/9389634/200279722-11882bf0-b7bc-4ba4-9b5c-5f627960fb94.png)

> The first step does not work either, because it always return "true".

What kind of trigger are you using and what is it supposed to be triggered by?

**Reply by ghost** [reply]
![image](https://user-images.githubusercontent.com/48225313/200293553-b55f050c-b0c3-4a99-a992-5b61ba36363b.png)
This was my first idea of updating an empty "Owner" field if an item is created or updated.

I don´t "update" data in the "relation table" because this is an create for mapping the current user to the empty "Owner" Field.

**Reply by br41nslug** [reply]
From the screenshot it looks you're not updating anything? 😬 You should either look into `filter` event instead of `action` or into using the `update data` as i think you want to be updating the `urmvitatility` table here.

**Reply by ghost** [reply]
Yes, because of the many unanswered questions, I'm not updating anything now, because I haven't managed it yet. Can we exchange directly and find a (paid) solution together? Maybe in a short video call meeting.

**Reply by ghost** [reply]
Okay .. start at zero .. I found a solution.

![image](https://user-images.githubusercontent.com/48225313/200405477-ac4198c3-8ec3-49ec-8946-4ba8218d9854.png)

I query the relation table for an item
`
{
    "filter": {
        "UrmVitality_id": {
            "_eq": "{{ $trigger.keys[0] }}"
        }
    }
}
`
And if there is no item found, I create one
`
{
    "UrmVitality_id": "{{ $trigger.keys[0] }}",
    "directus_users_id": {
        "id": "{{ $accountability.user }}"
    }
}
`

The real solution was rethinking. With Directus, you work closer to the actual data structure than with Strapi (where I´m coming from) and therefore you also only have to check and add to the relational table.

**Reply by timspi** [reply]
Maybe the reason why it was not working with the first approach is this open issue: https://github.com/directus/directus/issues/16127

When working with M2M fields, any updates made by flows to the payload are ignored at the moment.

**Reply by ghost** [reply]
Okay .. this seems to be a good reason. The new approach works fine but is never an generic approach for multiple collections. The first approach would work for "UrmVitality" and "UrmNodes" with the same "Owner" field in the same flow.
